mapie/regression/estimator.py

```diff
@@ -0,0 +1,504 @@
+        self.estimators_ = estimators_
+
+        return self
+
+    def _aggregate_with_mask(
+        self,
+        x: NDArray,
+        k: NDArray
+    ) -> NDArray:
+        """
+        Take the array of predictions, made by the refitted estimators,
+        on the testing set, and the 1-or-nan array indicating for each training
+        sample which one to integrate, and aggregate to produce phi-{t}(x_t)
+        for each training sample x_t.
+
+        Parameters:
+        -----------
+        x: ArrayLike of shape (n_samples_test, n_estimators)
+            Array of predictions, made by the refitted estimators,
+            for each sample of the testing set.
+
+        k: ArrayLike of shape (n_samples_training, n_estimators)
+            1-or-nan array: indicates whether to integrate the prediction
+            of a given estimator into the aggregation, for each training
+            sample.
+
+        Returns:
+        --------
+        ArrayLike of shape (n_samples_test,)
+            Array of aggregated predictions for each testing sample.
+        """
+        if self.method in self.no_agg_methods_ or self.cv in self.no_agg_cv_:
+            raise ValueError(
+                "There should not be aggregation of predictions "
+                f"if cv is in '{self.no_agg_cv_}' "
+                f"or if method is in '{self.no_agg_methods_}'."
+            )
+        elif self.agg_function == "median":
+            return phi2D(A=x, B=k, fun=lambda x: np.nanmedian(x, axis=1))
+        # To aggregate with mean() the aggregation coud be done
+        # with phi2D(A=x, B=k, fun=lambda x: np.nanmean(x, axis=1).
+        # However, phi2D contains a np.apply_along_axis loop which
+        # is much slower than the matrices multiplication that can
+        # be used to compute the means.
+        elif self.agg_function in ["mean", None]:
+            K = np.nan_to_num(k, nan=0.0)
+            return np.matmul(x, (K / (K.sum(axis=1, keepdims=True))).T)
+        else:
+            raise ValueError("The value of self.agg_function is not correct")
+
+    @staticmethod
+    def _predict_oof_estimator(
+        estimator: RegressorMixin,
+        X: ArrayLike,
+        val_index: ArrayLike,
+    ):
+        """
+        Perform predictions on a single out-of-fold model on a validation set.
+
+        Parameters
+        ----------
+        estimator: RegressorMixin
+            Estimator to train.
+
+        X: ArrayLike of shape (n_samples, n_features)
+            Input data.
+
+        val_index: ArrayLike of shape (n_samples_val)
+            Validation data indices.
+
+        Returns
+        -------
+        Tuple[NDArray, ArrayLike]
+            Predictions of estimator from val_index of X.
+        """
+        X_val = _safe_indexing(X, val_index)
+        if _num_samples(X_val) > 0:
+            y_pred = estimator.predict(X_val)
+        else:
+            y_pred = np.array([])
+        return y_pred, val_index
```

Comment:
UPD: move it to be close to `_fit_oof_estimator`. Better for reading.

Reply:
Done